context/src/main/java/io/grpc/Deadline.java

```diff
@@ -164,13 +164,19 @@ public int compareTo(Deadline that) {
     return 0;
   }
 
-  /** Time source representing nanoseconds since fixed but arbitrary point in time. */
-  abstract static class Ticker {
+  /**
+   * Time source representing nanoseconds since fixed but arbitrary point in time.
+   *
+   * <p>This is public for testing purposes.
+   */
+  public interface DeadlineTicker {
```

Comment:
Why not just call this Ticker and then people use Deadline.Ticker as the name? Let's not stutter unless we need to.